### PR TITLE
[Feature] Conditionable trait

### PR DESCRIPTION
**Pull request by maurohmartinez**
The PR for ti can be seen [here](https://github.com/Laravel-Backpack/CRUD/pull/5035).

### CODE REVIEW

File: 5.x/crud-fluent-syntax.md [hunk 1]
```diff
@@ -184,6 +184,20 @@ CRUD::field('products')
...
+Sometimes you may need a method to be chained on a field based on another condition. Backpack takes advantage of <a href="https://laravel.com/docs/10.x/queries#conditional-clauses" target="_blank">Laravel Conditionable</a> to easily let you accomplish that. You can either use ```->when()``` or ```unless()```:
+
+```php
+CRUD::field('email')
+    ->when($condition, function ($crud) {
+        $crud->unless($anothercondition, function ($crud) {
+            $crud->hint('My conditioned hint');
+        })
+    });
+```
```
Comment (tabacitu) [inline 5.x/crud-fluent-syntax.md:199]: Hmmm I think this example might be slightly misleading. 

1) What gets passed as a parameter to the function? Is the the `$crud` or is it actually the `$field`?!
2) Does it make more sense for us to teach people to use `->unless()` inside the closure... or outside it?

Please take another look at this @maurohmartinez - we want this example to be as clear as day.

File: 5.x/crud-fluent-syntax.md [hunk 2]
```diff
@@ -242,6 +256,20 @@ CRUD::column('name')->makeFirst();
...
+Sometimes you may need a method to be chained on a column based on another condition. Backpack takes advantage of <a href="https://laravel.com/docs/10.x/queries#conditional-clauses" target="_blank">Laravel Conditionable</a> to easily let you accomplish that. You can either use ```->when()``` or ```unless()```:
+```php
+CRUD::column('name')
+    ->unless($condition, function ($crud) {
+        $crud->wrapper([
+            'href' => function ($crud, $column, $entry, $related_key) {
+                return backpack_url('category/'.$related_key.'/show');
+            },
+        ])
+    });
```
Comment (tabacitu) [inline 5.x/crud-fluent-syntax.md:271]:
```suggestion
    ->unless($condition, function ($column) {
        $column->wrapper([
            'href' => function ($crud, $column, $entry, $related_key) {
                return backpack_url('category/'.$related_key.'/show');
            },
        ])
    });
```

File: 5.x/crud-fluent-syntax.md [hunk 3]
```diff
@@ -375,6 +403,21 @@ CRUD::button('name')->makeFirst();
...
+Sometimes you may need a method to be chained on a button based on another condition. Backpack takes advantage of <a href="https://laravel.com/docs/10.x/queries#conditional-clauses" target="_blank">Laravel Conditionable</a> to easily let you accomplish that. You can either use ```->when()``` or ```unless()```:
+
+```php
+CRUD::button('name')
+    ->unless($condition, function ($crud) {
+        $crud->before('description')
+            ->when($secondCondition, , function ($crud) {
+                $crud->forget('suffix');
+            });
+    });
```
Comment (tabacitu) [inline 5.x/crud-fluent-syntax.md:418]:
```suggestion
CRUD::button('name')
    ->unless($condition, function ($button) {
        $button->before('description')
            ->when($secondCondition, , function ($button) {
                $button->forget('suffix');
            });
    });
```

File: 5.x/crud-fluent-syntax.md [hunk 4]
```diff
@@ -565,6 +608,17 @@ CRUD::filter('name')->makeFirst();
...
+<a name="fluent-filters-conditionable-methods"></a>
+### Conditionable methods
+
+Sometimes you may need a method to be chained on a filter based on another condition. Backpack takes advantage of <a href="https://laravel.com/docs/10.x/queries#conditional-clauses" target="_blank">Laravel Conditionable</a> to easily let you accomplish that. You can either use ```->when()``` or ```unless()```:
+
+```php
+CRUD::filter('name')
+    ->when($condition, function ($crud) {
+        $crud->->remove();
+    });
```
Comment (tabacitu) [inline 5.x/crud-fluent-syntax.md:620]:
```suggestion
    ->when($condition, function ($filter) {
        $filter->remove();
    });
```

File: 5.x/crud-fluent-syntax.md (hunk 1 above)
Comment (maurohmartinez) [reply 5.x/crud-fluent-syntax.md]: I simplified it and renamed the `$crud` param to `$field`.

**Review by tabacitu — APPROVED** [review]

**Pull request merged**